build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -933,6 +933,13 @@ message SymlinkNode {
 message Digest {
   // The hash. In the case of SHA-256, it will always be a lowercase hex string
   // exactly 64 characters long.
+  //
+  // For any digest function having enumeration value 8 or higher, the hash MUST
+  // be prefixed with the shortest decimal representation of the enumeration
+  // value, followed by a colon. For example, for BLAKE3CONCAT, a hash may be
+  // "8:bb1d2c5731351d75e147129060313a04c9cb59edb7ffda0873470f891198cf2b". This
+  // prevents ambiguities between digest functions generating hashes of equal
+  // length.
```

Comment:
Why not use a separate field for the hash type, instead of a prefix in the hash string? Is it some kind of size optimisation (IIRC @bergsieker mentioned something about this in the call), or is it just so we have a way to specify the hash type in bytestream ReadRequest and WriteRequest without forking the bytestream protocol?

Reply:
Maybe the reason is to support the existing ByteStream uri encoding.

Reply:
Exactly; it's defined like this, so that we don't need to make any modifications to the URL scheme supported by the ByteStream API. Both as Digest messages and the ByteStream API will we use the "8:" prefix to signify BLAKE3CONCAT.

Reply:
Is it worth using a field here, and then specifying where we describe the bytestream resource format that the prefix is needed? That way we could defer the bytestream decision (and then maybe reconsider forking/extending bytestream).

Reply:
That solution would also work, but would make life harder for clients that are currently completely oblivious of checksumming. Those may currently depend on constructing ByteStream URLs based on what's in the Digest messages and vice versa. That ship might have sailed at this point.

Reply:
If they need to support multiple hash functions with the same hash length in a single instance, yes, they would have to support the ByteStream addition. As discussed in the meeting, such a requirement depends on how the migration between hash functions is handled. It's not needed in all cases.

My main point is that if we already know that at least some implementations require changes either way, the benefit of a hack such as encoding the hash function number into a string (changing the format of an existing field) is limited, and a cleaner solution is preferable, in my opinion.